### ISSUE TITLE
General questions

### ISSUE THREAD

**Issue by ctchavez6** [PyAbel/PyAbel#353]
I have an image I am trying to inverse transform here: https://github.com/ctchavez6/csv_test_files/blob/main/phi_minus_background_circ.png
![phi_minus_background_circ](https://user-images.githubusercontent.com/86802224/163479999-e294716f-87f9-45e4-8862-36388c929b57.png)
(sorry I'm not sure which is better)
I think I have the orientation correct although its not perfectly aligned: the axis of the transform is at some angle to the axis of the object. How would this affect the transform? Second, down the axis of the transform is a line of high intensity that I'm not sure where it came from. Is this the correct use of the inverse Abel? 
I'm just confident enough to believe the result but skeptical enough to question it.

**Comment by DanHickstein** [comment]
PyAbel makes the assumption that axis of symmetry runs along the vertical dimension of your input array. So, if your image isn't aligned, then the transform will look strange. 

You could try using [scipy.ndimage.rotate](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.rotate.html) to rotate your image to the correct orientation before transforming.

**Comment by ctchavez6** [comment]
Thank you for your help, I have rotated the image like you suggested but I am still getting discontinuity near the transform axis. The first image is the raw data, followed by the hansenlaw inverse, and last is the basex inverse transform.
![ableinverse](https://user-images.githubusercontent.com/86802224/163635960-d06f9e78-2252-4bd8-93da-2bfc72c7bdf2.PNG)

**Comment by DanHickstein** [comment]
For real-world data, it's typical to have a lot of noise near the centerline. If you adjust the limits to look at the rest of the image, how does it look?

**Comment by ctchavez6** [comment]
![phi_minus_background](https://user-images.githubusercontent.com/86802224/163838523-b8b720e7-dc87-4c96-934e-c43a2a9ebbc2.png)
Square photo of a circular laser beam. Very noisy outside of the beam. I forced the edges to zero this way to help the boundary condition of zero at the edges.

**Comment by DanHickstein** [comment]
Yes, I think that it makes sense to force the noisy region to zero. 

I was asking about the transformed image. I'm wondering if your signal is actually there, but it's just at a level far below the centerline noise.

**Comment by ctchavez6** [comment]
Ah, yes, it does appear that there is a region with the correct signal. here is an image limited to -2 and 4. I am also concerned that I am seeing negative values
![limited_jet](https://user-images.githubusercontent.com/86802224/163858308-f7046643-6399-4c78-8755-6f2363a18314.PNG)
.

**Comment by ctchavez6** [comment]
# -*- coding: utf-8 -*-

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import os.path
from scipy import ndimage as nd
import numpy as np
import matplotlib.pyplot as plt
import abel
import cv2 as cv
# This example demonstrates a BASEX transform of an image obtained using a
# velocity map imaging (VMI) photoelecton spectrometer to record the
# photoelectron angualar distribution resulting from above threshold ionization
# (ATI) in xenon gas using a ~40 femtosecond, 800 nm laser pulse.
# This spectrum was recorded in 2012 in the Kapteyn-Murnane research group at
# JILA / The University of Colorado at Boulder
# by Dan Hickstein and co-workers (contact DanHickstein@gmail.com)
# https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.109.073004
#
# Before you start your own transform, identify the central pixel of the image.
# It's nice to use a program like ImageJ for this.
# https://imagej.nih.gov/ij/

# Specify the path to the file
filename = 'D:/1DataAnalysis/April_first/Timing/14/phi_minus_background_circ.png'

# Step 1: Load an image file as a numpy array
print('Loading ' + filename)
raw_data = plt.imread(filename).astype('float64')

src = cv.imread(cv.samples.findFile(filename), cv.IMREAD_COLOR)
src = nd.rotate(src,4.35)
src_gray = cv.cvtColor(src, cv.COLOR_BGR2GRAY)  # converts to grayscale

# Step 2: Specify the origin in (row, col) format
# origin = (1,170)
# or, use automatic centering
origin = 'com'
# origin = 'gaussian'

# Step 3: perform the BASEX transform!


recon = abel.Transform(src_gray, direction='inverse', method='basex').transform
print(np.shape(recon))
new = np.where(recon[:][:]<4,recon[:][:],4)
new = np.where(new[:][:]>-2,new[:][:],-2)


recon2 = abel.Transform(src_gray, direction='inverse', method = 'hansenlaw').transform


fig = make_subplots(
    rows=1, cols=3,
    specs=[[{'type': 'surface'}, {'type': 'surface'}, {'type': 'surface'}]])

fig.add_trace(
    go.Surface(z = src_gray, colorscale='Viridis', showscale=False),
    row=1, col=1)


fig.add_trace(
    go.Surface(z = new, colorscale='Viridis', showscale=False),
    row=1, col=2)

fig.add_trace(
    go.Surface(z = recon2, colorscale='Viridis', showscale=False),
    row=1, col=3)

fig.show()
print(np.max(new),np.min(new))

**Comment by DanHickstein** [comment]
Okay, that looks somewhat like an Abel transform of your input. Because of noise, you can expect to see negative values.

**Comment by ctchavez6** [comment]
Awesome, Thank you for your help. Ill work on smoothing the transform so its more continuous at the center.

**Comment by DanHickstein** [comment]
Try playing with the `reg` parameter in basex. 

So, change your line: 
`recon = abel.Transform(src_gray, direction='inverse', method='basex').transform`
to
`recon = abel.Transform(src_gray, direction='inverse', method='basex', transform_options=dict(reg=10)).transform`

Then play around with the regularization parameter until you get the level of smoothing/center-line-noise-suppression that you want.

**Comment by ctchavez6** [comment]
Wow, that does help a lot! i need to use reg = ~10000 which i read is magnitudes bigger than what is normal but i think it might be justified given the noise to signal ratio.

**Comment by DanHickstein** [comment]
Awesome! Glad that it helped. If you have a chance, post an image of your result with the regularization factor. 

And yeah, it's not surprising to me that you benefit from a larger reg parameter than typical, since your features are larger (in pixels) than the "typical" photoelectron spectroscopy image.

**Comment by ctchavez6** [comment]
![14_INVERSE_TRANSFORM](https://user-images.githubusercontent.com/86802224/164037667-e87274ab-c15e-41d0-ab58-51eecbfca738.PNG)
Here it is close to its final form. Adjusted the center a bit more and rotated in imageJ before the transform.
